czbiohub/imagingDB
# Add image hash to the database

# Description
As discussed in #37, this PR adds an image hash to the imagingDB. Upon uploading a sha256 has is calculated. When images are uploaded frame-by-frame, a hash is generated for each frame (calculated from the actual image data, not the file) and added to the sha256 column in the frame table. In the file upload case, the hash is generated from the whole file being uploaded and added to the sha256 column of the file_global table. Closes #37.

# Testing
I am currently working on the tests, but my plan is to do the following:
- add test to `file_splitter_tests.py` for the hash function (make sure the hash of the test image is correct)
- add test to make sure downloaded frames and downloaded files have the correct hash (see discussion below)
